src/wallet/scriptpubkeyman.cpp

```diff
@@ -357,7 +357,7 @@ void LegacyScriptPubKeyMan::MarkUnusedAddresses(const CScript& script)
         if (it != mapKeyMetadata.end()){
             CKeyMetadata meta = it->second;
             if (!meta.hd_seed_id.IsNull() && meta.hd_seed_id != m_hd_chain.seed_id) {
-                bool internal = (meta.key_origin.path[1] & ~BIP32_HARDENED_KEY_LIMIT) != 0;
+                bool internal = ((meta.key_origin.path[1] == 0x80000003) || (meta.key_origin.path[1] == 0x80000000)) ? 0 : 1; 
```

Comment:
For a followup (again not to block this merge), is it worth defining named constants to represent these values? They're repeated a couple of times, so they just about reach the threshold where it's easier to change one spot and refer to them symbolically.

Reply:
Yes, also needs docs. See https://github.com/dogecoin/dogecoin/pull/2977#discussion_r928279183 - I'll do a followup PR after this.

Reply:
@chromatic I have attempted to improve this in #3128